BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1668 @@
+--- a/api/pom.xml
++++ b/api/pom.xml
+@@ -18,6 +18,28 @@
+     <name>Waterfall-API</name>
+     <description>API implemented by the Elastic Portal Suite</description>
+ 
++    <!-- Waterfall start - add snapshot repository for Adventure's BungeeCord serializer and add BOM -->
++    <repositories>
++        <!-- Waterfall - Add repository required for Adventure's BungeeCord serializer -->
++        <repository>
++            <id>sonatype-oss-snapshots</id>
++            <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
++        </repository>
++    </repositories>
++
++    <dependencyManagement>
++        <dependencies>
++            <dependency>
++                <groupId>net.kyori</groupId>
++                <artifactId>adventure-bom</artifactId>
++                <version>4.7.0</version>
++                <type>pom</type>
++                <scope>import</scope>
++            </dependency>
++        </dependencies>
++    </dependencyManagement>
++    <!-- Waterfall end -->
++
+     <dependencies>
+         <dependency>
+             <groupId>io.github.waterfallmc</groupId>
+@@ -82,5 +104,24 @@
+             <artifactId>slf4j-api</artifactId>
+             <version>1.7.25</version>
+         </dependency>
++        <!-- Waterfall start - Add Adventure -->
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-api</artifactId>
++        </dependency>
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-gson</artifactId>
++        </dependency>
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-legacy</artifactId>
++        </dependency>
++        <dependency>
++            <groupId>net.kyori</groupId>
++            <artifactId>adventure-text-serializer-bungeecord</artifactId>
++            <version>4.0.0-SNAPSHOT</version>
++        </dependency>
++        <!-- Waterfall end -->
+     </dependencies>
+ </project>
+diff --git a/api/src/main/java/net/md_5/bungee/api/CommandSender.java b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+index d5888311..72f3da4c 100644
+--- a/api/src/main/java/net/md_5/bungee/api/CommandSender.java
++++ b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+@@ -3,7 +3,7 @@ package net.md_5.bungee.api;
+ import java.util.Collection;
+ import net.md_5.bungee.api.chat.BaseComponent;
+ 
+-public interface CommandSender
++public interface CommandSender extends net.kyori.adventure.audience.Audience // Waterfall - we're going on an adventure!
+ {
+ 
+     /**
+@@ -34,14 +34,18 @@ public interface CommandSender
+      * Send a message to this sender.
+      *
+      * @param message the message to send
++     * @deprecated in favour of {@link net.kyori.adventure.audience.Audience#sendMessage(net.kyori.adventure.text.Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
```

Comment:
You can simplify this to just `// Waterfall`, same applies to *all* other comments like this.

Reply:
I only added the description to indicate which patch this was changed in, but alrighty then, I'll do this.